### PR TITLE
doc: develop: tools: add VS Code setup guide

### PR DESCRIPTION
**Pull request by deribaucourt**
This basic guide only covers linting so far. It could be expanded in the future to be more accurate and detailed. The current configuration allows to navigate the code but some symbols are not recognized by the linter, and the native compiler is always used.

### CODE REVIEW

**Comment by kartben** [comment]
can you submit as an actual Zephyr PR? This looks like it's already in a very decent shape :)

File: doc/develop/tools/vscode.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+VS Code
+#######
+
+`Visual Studio Code`_ is a popular cross-platform IDE that supports C projects and has a rich set of
+extensions.
+
+This guide describes the process of setting up VS Code for Zephyr's
+:zephyr:code-sample:`blinky` sample in VS Code.
+
+The instructions have been tested on Linux. In terms of the CLion workflow, the steps would be the
```
Comment (kartben) [inline doc/develop/tools/vscode.rst:12]: clion?

File: doc/develop/tools/vscode.rst [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+generation of this file through the following configuration:
+
+#. Open your projects' CMakeLists.txt file and add the following line.
+   In our case, we will add it to the :file:`samples/basic/blinky/CMakeLists.txt` file.
+   ``set(CMAKE_EXPORT_COMPILE_COMMANDS ON)``
+
+#. Build the project to generate the `compile_commands.json` file:
+   ``west build -b native_posix_64 samples/basic/blinky``
+
+Configure linting
```
Comment (kartben) [inline doc/develop/tools/vscode.rst:58]: is this really called "just" linting btw? given this also enables code navigation, i would have thought this has maybe another name?

File: doc/develop/tools/vscode.rst [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+#. Navigate to your Zephyr workspace (i.e.the :file:`zephyrproject` folder in your HOME directory if
+   you have followed the Getting Started instructions), then select
+   :file:`zephyr/samples/basic/blinky` or another sample project folder.
+
+#. If prompted, enable workspace trust.
+
+Generate compile commands
+*************************
+
+#The setup showcased in this document relies on the `compile_commands.json` file generated by CMake
```
Comment (kartben) [inline doc/develop/tools/vscode.rst:47]:
```suggestion
The setup showcased in this document relies on the `compile_commands.json` file generated by CMake
```

**Comment by deribaucourt** [comment]
Thanks a lot for your input @kartben 
This is a WIP draft. I'll implement your suggestions and submit it as an official PR once done.

File: doc/develop/tools/vscode.rst (hunk 1 above)
Comment (deribaucourt) [reply doc/develop/tools/vscode.rst:12]: I based the document on the CLion guide, then my colleague @lucbeaufils tested the instructions from scratch.
Fixed the forgotten ref

File: doc/develop/tools/vscode.rst (hunk 2 above)
Comment (deribaucourt) [reply doc/develop/tools/vscode.rst:58]: Rephrased to "Configure the C/C++ extension" which is more accurate.

File: doc/develop/tools/vscode.rst (hunk 3 above)
Comment (deribaucourt) [reply doc/develop/tools/vscode.rst:47]: Done

**Comment by deribaucourt** [comment]
Submitted to Zephyr: https://github.com/zephyrproject-rtos/zephyr/pull/74259

File: doc/develop/tools/vscode.rst (hunk 2 above)
Comment (kartben) [reply doc/develop/tools/vscode.rst:58]: @deribaucourt please make sure the PR you just made to Zephyr actually has the fixes you said you have applied :)

Comment (deribaucourt) [reply doc/develop/tools/vscode.rst:58]: Thanks, looks like I miss-managed my git branches ><

**Pull request closed without merging**